### PR TITLE
#9Task. Elizaveta Panasiuk

### PR DESCRIPTION
**Pull request by ElizavetaPanasiuk**
https://travis-ci.com/github/ElizavetaPanasiuk/js-assignments/builds/175862075

### CODE REVIEW

File: task/09-functions-n-closures-tasks.js [hunk 1]
```diff
@@ -26,7 +26,9 @@
  *
  */
 function getComposition(f,g) {
-    throw new Error('Not implemented');
+    return function(x) {
+        return f(g(x));
+      }
 }
```
Comment (elizavetaShcherbatykhAis) [inline task/09-functions-n-closures-tasks.js:30]: You can write it like:
return (x) => f(g(x));

File: task/09-functions-n-closures-tasks.js [hunk 2]
```diff
@@ -47,7 +49,9 @@ function getComposition(f,g) {
  *
  */
 function getPowerFunction(exponent) {
-    throw new Error('Not implemented');
+    return function (x) {
+        return Math.pow(x, exponent);
+    }
```
Comment (elizavetaShcherbatykhAis) [inline task/09-functions-n-closures-tasks.js]: Same :)

Comment (ElizavetaPanasiuk) [reply task/09-functions-n-closures-tasks.js:54]: What about marking this PR as draft?

File: task/09-functions-n-closures-tasks.js [hunk 3]
```diff
@@ -84,7 +92,19 @@ function getPolynom() {
...
+        
+        if (!memory.has(arg)) {
+        result = func(arg);
+        memory.set(arg, result);
+        }else {
+        result = memory.get(arg);
+        } 
+
+        return result;
+    };
```
Comment (elizavetaShcherbatykhAis) [inline task/09-functions-n-closures-tasks.js]: You can do it much easier in 1-2 rows :)

File: task/09-functions-n-closures-tasks.js [hunk 4]
```diff
@@ -149,8 +188,8 @@ function logger(func, logFunc) {
  *   partialUsingArguments(fn, 'a','b','c')('d') => 'abcd'
  *   partialUsingArguments(fn, 'a','b','c','d')() => 'abcd'
  */
-function partialUsingArguments(fn) {
-    throw new Error('Not implemented');
+function partialUsingArguments(fn, ...args) {
```
Comment (elizavetaShcherbatykhAis) [inline task/09-functions-n-closures-tasks.js]: You can't change function arguments, sorry

**Comment by elizavetaShcherbatykhAis** [comment]
Please, check the comments and try to improve your code.
I'll mark the PR as draft please click "ready for review" when it will be finished!
Thank you!

**Comment by elizavetaShcherbatykhAis** [comment]
Cool! I close PR as accepted

**Pull request closed without merging**